### PR TITLE
Adding dependency parsing in optimizer

### PR DESCRIPTION
**Pull request by pritom5**

### CODE REVIEW

File: owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        rock=df.getOWLNamedIndividual("rock", pm);
+
+        nrOfInhabitants = df.getOWLDataProperty("nrOfInhabitants", pm);
+        dataRange = df.getOWLDatatypeMinInclusiveRestriction(10000000);
+
+        ToStringRenderer.getInstance().setRenderer(new DLSyntaxObjectRenderer());
+    }
+
+    @Test
+    public void testComple() {
```
Comment (AmbujMondal) [inline owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java]: I guess you want to name it testCompile. please check.

File: owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+            List<IndexedWord> finalTextList=new ArrayList<>();
+            if(combinedCcCommaList.size()>=1) {
+                subjectsChecker = checkAllSubjectsSame(nodeList, verbIndex, combinedCcCommaIndex);
+                verbsChecker = checkAllVerbsSame(verbList);
+            }
+            if(combinedCcCommaList.size()>=1 && verbList.size()==combinedCcCommaList.size()+1) {
+                //System.out.println("HHHHHHHHHHHHHH");
+                if (subjectsChecker) {
+                    //System.out.println("HHHHHHHHHHHHHH");
+                    //System.out.println(dependencyParse);
```
Comment (AmbujMondal) [inline owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java]: Remove unwanted commented code.

File: owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+            stanfordCoreNLP.annotate(coreDocument);
+            CoreSentence sentence = coreDocument.sentences().get(0);
+            SemanticGraph dependencyParse = sentence.dependencyParse();
+
+
+
+            List<IndexedWord> nodeList = dependencyParse.vertexListSorted();
+            //System.out.println("****************** TEST *********************");
+            //System.out.println("List: " + nodeList);
+            //System.out.println("List: " + nodeList.get(0).value());
```
Comment (AmbujMondal) [inline owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java]: cleanup unwanted commented code.

**Review by AmbujMondal — CHANGES_REQUESTED** [review]
Nice work 👍🏼

File: owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java (hunk 1 above)
Comment (pritom5) [reply owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java]: corrected the test case name

File: owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java (hunk 3 above)
Comment (pritom5) [reply owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java]: done

File: owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        rock=df.getOWLNamedIndividual("rock", pm);
+
+        nrOfInhabitants = df.getOWLDataProperty("nrOfInhabitants", pm);
+        dataRange = df.getOWLDatatypeMinInclusiveRestriction(10000000);
+
+        ToStringRenderer.getInstance().setRenderer(new DLSyntaxObjectRenderer());
+    }
+
+    @Test
+    public void testSubjectAndVerAggregation() {
```
Comment (AparnaSivaprasad) [inline owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java]: Spelling mistake in test case name

File: owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+        // a person that sings karaoke or a person that sings  jazz
+        ce = df.getOWLObjectUnionOf(df.getOWLObjectIntersectionOf(df.getOWLObjectHasValue(sings, karaoke), man), df.getOWLObjectIntersectionOf(df.getOWLObjectHasValue(sings, Jazz), man));
```
Comment (AparnaSivaprasad) [inline owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java]: It'll be more readable if you split this into multiple lines

File: owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+                System.out.println("Text after our changes : " + finalText);
+            }
+
+
+            return finalText;
+        } catch (Exception e) {
+            return text;
+        }
+    }
+    //method to check whether all verbs in the sentence are same or not
```
Comment (AparnaSivaprasad) [inline owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java]: Please add a blank line before each new method.
Also, if possible, please start each comment with a capital letter.

File: owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+            stanfordCoreNLP.annotate(coreDocument);
+            CoreSentence sentence = coreDocument.sentences().get(0);
+            SemanticGraph dependencyParse = sentence.dependencyParse();
+
+
+
+            List<IndexedWord> nodeList = dependencyParse.vertexListSorted();
+
+            //initializing
+            System.out.println("Example: dependency parse");
```
Comment (AparnaSivaprasad) [inline owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java]: Do we need these many print statements?

File: owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java [hunk 8]
```diff
@@ -0,0 +1,239 @@
+package org.aksw.owl2nl;
+
+import edu.stanford.nlp.ling.IndexedWord;
+import edu.stanford.nlp.pipeline.CoreDocument;
+import edu.stanford.nlp.pipeline.CoreSentence;
+import edu.stanford.nlp.pipeline.StanfordCoreNLP;
+import edu.stanford.nlp.semgraph.SemanticGraph;
+import edu.stanford.nlp.semgraph.SemanticGraphFormatter;
+//import org.graalvm.compiler.core.common.type.ArithmeticOpTable;
```
Comment (AparnaSivaprasad) [inline owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java]: Are you keeping this line commented for testing in the future?
If so, please specify that as a comment above it.

File: owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+import java.util.List;
+import java.util.Properties;
+
+public class OptimizerDepParse {
+
+    public String optimize(String text) {
+        try {
+            if (text == null || text == "") return text;
+
+            //text="something that a man that sings rock and that a man that sings jazz or a man that sings karaoke  or a man that sings bal";
```
Comment (AparnaSivaprasad) [inline owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java]: Same as the previous comment.

File: owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java (hunk 4 above)
Comment (pritom5) [reply owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java]: done

File: owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java (hunk 5 above)
Comment (pritom5) [reply owl2nl/src/test/java/org/aksw/owl2nl/OptimizerDepParseTest.java]: Thank you. done

File: owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java (hunk 6 above)
Comment (pritom5) [reply owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java]: Done

File: owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java (hunk 7 above)
Comment (pritom5) [reply owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java]: removed some print statements but only kept the star printing for separation of test cases.

File: owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java (hunk 9 above)
Comment (pritom5) [reply owl2nl/src/main/java/org/aksw/owl2nl/OptimizerDepParse.java]: For now need it for demonstration purposes. I will remove it afterwards. Thank you

**Pull request closed without merging**